### ISSUE TITLE
Case sensivity in `show` command

### ISSUE THREAD

**Issue by moteus** [keplerproject/luarocks#633]
show command works only for low case e.g. for LuaSocket module.
Rockspec define `package = "LuaSocket"`
I can install it like `luarocks install LuaSocket`
When I install it I get `luasocket 3.0rc1-2 is now installed in c:\luarocks\x86\5.1\systree (license: MIT)`
When I call `luarocks show luasocket` I get `LuaSocket 3.0rc1-2 - Network support for the Lua language`
But wen I call `luarocks show LuaSocket` i get `Error: cannot find package LuaSocket`

I think LuaRocks should define some rules about such names.

**Comment by mpeterv** [comment]
I think that internally luarocks uses only lower case names, So `package = "LuaSocket"` and `package = "luasocket"` are equivalent except for what `show` displays. Some commands (install, build, list, search) convert user provided name to lower, some commands (download, show, doc, remove) don't. I think that all commands should convert to lowercase.

**Comment by hishamhm** [comment]
> I think LuaRocks should define some rules about such names.

There are some rules in place: .rock and .rockspec files must be lowercase-only. The package field inside the rockspec contains the name with the user's desired display case (though things like `luarocks list` use the lowercase name, which I think is fine — the display names are used at luarocks.org for example).

> I think that all commands should convert to lowercase.

Agreed. Apparently, I made the original set of commands case-insensitive, but haven't followed up with this practice in the ones I added later. This can be fixed.

**Comment by hishamhm** [comment]
> I think LuaRocks should define some rules about such names.

There are some rules in place: .rock and .rockspec files must be lowercase-only. The package field inside the rockspec contains the name with the user's desired display case (though things like `luarocks list` use the lowercase name, which I think is fine — the display names are used at luarocks.org for example).

> I think that all commands should convert to lowercase.

Agreed. Apparently, I made the original set of commands case-insensitive, but haven't followed up with this practice in the ones I added later. This can be fixed.
